packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -725,3 +725,101 @@ class _MotionEventsDispatcher {
   bool isSinglePointerAction(PointerEvent event) =>
       !(event is PointerDownEvent) && !(event is PointerUpEvent);
 }
+
+/// Implement this if you need to ask [PlatformViewRenderBox] to create a custom layer instead of a standard [PlatformViewLayer].
+///
+/// The [renderBox] is the render object whose [paint] method is invoked to create this layer.
+/// The [context] and [offset] are the two parameters you would normally get in the `RenderBox.paint` method.
+/// The [id] is the platform view is provided to you by the framework.
+typedef PlatformViewRenderBoxLayerFactory = Layer Function(PlatformViewRenderBox renderBox, PaintingContext context, Offset offset, int id);
+
+/// The render object for `PlatformViewSurface`.
+///
+/// It is responsible for painting, gestures and semantics.
+class PlatformViewRenderBox extends RenderBox {
+
+  /// Creating the render object for `PlatformViewSurface`.
+  ///
+  /// [controller], [id] must not be null.
+  /// Optionally, you can pass a [customLayerFactory] to generate a custom layer in the [paint] method.
+  /// A `PlatformViewLayer` is used by default.
+  PlatformViewRenderBox({
+    @required PlatformViewController controller,
+    @required int id,
+    PlatformViewRenderBoxLayerFactory customLayerFactory,
```

Comment:
nit: I would default this to something and assert that it's not null, and then your paint method doesn't have to do the null check every time.

Reply:
Seeing it this way, maybe we just want to call it `layerBuilder`?